bite-size-notes/week-4-e-stochastic bellman equation.md

```diff
@@ -0,0 +1,167 @@
+$$
+
+and for $k = 0, . . . , N−1$, let
+
+$$
+  V_k^{\*} = \min_{u_k \in U_k (x_k)} \mathbb{E} \[ g_k(x_k, u_k, w_k) + V^*_{k+1}(f_k(x_k, u_k, w_k)) \]
+$$
+
+- If $u^∗_k = \mu^∗_k(x_k)$ minimizes the right side of this equation for each $x_k$ and $k$, the policy ${\pi}^{∗} = {\mu_0^{\*}, . . . , \mu_{N-1}^\*}$ is optimal.
+
+$$
+    {\mu_k} (x_k) \in arg\min_{u_k \in U_k (x_k)} \mathbb{E} \[ g_k(x_k, u_k, w_k) + V^*_{k+1}(f_k(x_k, u_k, w_k))
+$$
+
+- For finding suboptimal policy, approximate the cost-go-function $\bar{V}^*_{k}$(using Model Approximation, Neural-Nets, Rollout algorithm, Model Predictive Control..etc)
+
+$$
+   \bar{V_k}^{\*} = \min_{u_k \in U_k (x_k)} \mathbb{E} \[ g_k(x_k, u_k, w_k) + \bar{V}^*_{k+1}(f_k(x_k, u_k, w_k)) \
+$$
+
+where suboptimal control policy obtained 
+
+$$
+    \bar{\mu_k} (x_k) \in arg\min_{u_k \in U_k (x_k)} \mathbb{E} \[ g_k(x_k, u_k, w_k) + \bar{V}^*_{k+1}(f_k(x_k, u_k, w_k))
+$$
+
+
+The key fact is that for every initial state $x_0$, the optimal cost $J^\*(x_0)$ is equal to the function $J_0^\*(x_0)$, obtained at the __last step__ of the above DP algorithm.
+
+
+
+However, the DP algorithm can be very time-consuming, but more than the deterministic case since it involves the expected value operation and a rapid increase of the required computation and memory storage as the problem’s size increases. This difficulties motivate suboptimal control schemes, such as approximation in value space whereby we replace $J_k^\*$ with easier obtainable approximation $\bar{J}_k$. $\bar{J}_k$ should be “close” to the optimal cost-to-go function $J_k^\*$ for all $k$.
+schemes. From the looking ahead minimization of cost function, we obtain the suboptimal(heuristic) policies $\bar{\mu}$ which are used in situations where finding an optimal solution is computationally intractable or impractical. 
+
+## Conclusion
+During the execution of a controlled system, at any point in time the system trajectory may be split in two. 
+- The optimal future trajectory is independent of the past (Markov Assumption), as the system’s state captures the full history of what occurred, and thus each trajectory segment is itself optimal (Bellman's Optimality). 
+- Goal of Optimal Control in stocahstic horizon is to minimize expected value of the sum of costs, subject to dynamics and other constraints and disturbances.
+- Optimal control are not deterministic and the curse of dimensionality, approximation of value space and searching the suboptimal policy is the main objective of stochastic dynamic programming.
+
+## References
+1. Karen Leung, "Bellman Equation & LQR lecture", 2024 4/15~4/17
+2. Matanya B. Horowitz, "Efficient Methods for Stochastic Optimal Control", 2014.
+3. Dimitri P. Bertsekas, "Reinforcement Learning and Optimal Control", Athena Scientific, Belmont, Massachusetts, 2019.
+4. Dimitri P. Bertsekas, "Dynamic Programming and Suboptimal Control: A survey from ADP to MPC", European Journal of Control, 2005.
+

```

Comment:
1. I like the conclusion that conclude the interpretation about the optimal stochastic system, that we are optimizing the expect value, and an algorithm. 

2. Can use bullet points is the difference part to list the difference.

Reply:
Thanks